File: 파이썬/2.검색/7795.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/7795
'''

def down(arr, length, target) :
    low = 0
    high = length - 1
    while low < high :
        mid = (low + high) // 2
        if arr[mid] >= target :
            high = mid
        else :
            low = mid + 1
    return high

def up(arr, length, target) :
    low = 0
    high = length - 1
    while low < high :
        mid = (low + high) // 2
        if arr[mid] > target :
            high = mid
        else :
            low = mid + 1
    return high

n = int(input())
a = list(map(int, input().split()))
m = int(input())
b = list(map(int, input().split()))

a.sort()

for i in range(m) :
    r1 = down(a, n, b[i])
    r2 = up(a, n, b[i])
    if r2 == n-1 and a[n-1] == b[i] :
        r2+=1
    print(r2 - r1, end = ' ')

File: C++/28. 이진 검색 알고리즘 - 중급/7795.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/7795
*/

#include <iostream>
#include <algorithm>

using namespace std;

int A[20000];
int B[20000];

int binarySearch(int len, int target){
    int low = 0;
    int high = len - 1;
    int mid;
    while(low <= high){
        mid = (low + high) / 2;
        if(B[mid] == target){
            high = mid - 1;
        }else if(B[mid] < target){
            low = mid + 1;
        }else{
            high = mid - 1;
        }
    }
    return low;
}

int main(){
    int t;
    cin >> t;
    for(int i=0; i<t; i++){
        int n, m;
        cin >> n >> m;
        for(int j=0; j<n; j++){
            cin >> A[j];
        }
        for(int j=0; j<m; j++){
            cin >> B[j];
        }
        sort(B, B+m);
        int sum = 0;
        for(int j=0; j<n; j++){
            sum += binarySearch(m, A[j]);
        }
        cout << sum << "\n";
    }
    
    return 0;
}